### PR TITLE
feat: support `DefineStmt` (aggregate)

### PR DESCRIPTION
**Pull request by cvng**
## What kind of change does this PR introduce?

feat: support `DefineStmt` (aggregate): `create aggregate ...`

## What is the current behavior?

Parser panics

## What is the new behavior?

```rust
pg_query_root: Some(
        DefineStmt(
            DefineStmt {
                kind: ObjectAggregate,
                oldstyle: false,
                defnames: [
                    Node {
                        node: Some(
                            String(
                                String {
                                    sval: "percentile_disc",
                                },
                            ),
                        ),
                    },
                ],
                args: [
                    Node {
                        node: Some(
                            List(
                                List {
                                    items: [
                                        Node {
                                            node: Some(
                                                FunctionParameter(
                                                    FunctionParameter {
                                                        name: "",
                                                        arg_type: Some(
                                                            TypeName {
                                                                names: [
                                                                    Node {
                                                                        node: Some(
                                                                            String(
                                                                                String {
                                                                                    sval: "float8",
                                                                                },
                                                                            ),
                                                                        ),
                                                                    },
                                                                ],
                                                                type_oid: 0,
                                                                setof: false,
                                                                pct_type: false,
                                                                typmods: [],
                                                                typemod: -1,
                                                                array_bounds: [],
                                                                location: 45,
                                                            },
                                                        ),
                                                        mode: FuncParamDefault,
                                                        defexpr: None,
                                                    },
                                                ),
                                            ),
                                        },
                                        Node {
                                            node: Some(
                                                FunctionParameter(
                                                    FunctionParameter {
                                                        name: "",
                                                        arg_type: Some(
                                                            TypeName {
                                                                names: [
                                                                    Node {
                                                                        node: Some(
                                                                            String(
                                                                                String {
                                                                                    sval: "anyelement", // appears once
                                                                                },
                                                                            ),
                                                                        ),
                                                                    },
                                                                ],
                                                                type_oid: 0,
                                                                setof: false,
                                                                pct_type: false,
                                                                typmods: [],
                                                                typemod: -1,
                                                                array_bounds: [],
                                                                location: 61,
                                                            },
                                                        ),
                                                        mode: FuncParamDefault,
                                                        defexpr: None,
                                                    },
                                                ),
                                            ),
                                        },
                                    ],
                                },
                            ),
                        ),
                    },
                    Node {
                        node: Some(
                            Integer(
                                Integer {
                                    ival: 1,
                                },
                            ),
                        ),
                    },
                ],
                definition: [
                    Node {
                        node: Some(
                            DefElem(
                                DefElem {
                                    defnamespace: "",
                                    defname: "sfunc",
                                    arg: Some(
                                        Node {
                                            node: Some(
                                                TypeName(
                                                    TypeName {
                                                        names: [
                                                            Node {
                                                                node: Some(
                                                                    String(
                                                                        String {
                                                                            sval: "ordered_set_transition",
                                                                        },
                                                                    ),
                                                                ),
                                                            },
                                                        ],
                                                        type_oid: 0,
                                                        setof: false,
                                                        pct_type: false,
                                                        typmods: [],
                                                        typemod: -1,
                                                        array_bounds: [],
                                                        location: 99,
                                                    },
                                                ),
                                            ),
                                        },
                                    ),
                                    defaction: DefelemUnspec,
                                    location: 91,
                                },
                            ),
                        ),
                    },
                    Node {
                        node: Some(
                            DefElem(
                                DefElem {
                                    defnamespace: "",
                                    defname: "stype",
                                    arg: Some(
                                        Node {
                                            node: Some(
                                                TypeName(
                                                    TypeName {
                                                        names: [
                                                            Node {
                                                                node: Some(
                                                                    String(
                                                                        String {
                                                                            sval: "internal",
                                                                        },
                                                                    ),
                                                                ),
                                                            },
                                                        ],
                                                        type_oid: 0,
                                                        setof: false,
                                                        pct_type: false,
                                                        typmods: [],
                                                        typemod: -1,
                                                        array_bounds: [],
                                                        location: 147,
                                                    },
                                                ),
                                            ),
                                        },
                                    ),
                                    defaction: DefelemUnspec,
                                    location: 139,
                                },
                            ),
                        ),
                    },
                    Node {
                        node: Some(
                            DefElem(
                                DefElem {
                                    defnamespace: "",
                                    defname: "finalfunc",
                                    arg: Some(
                                        Node {
                                            node: Some(
                                                TypeName(
                                                    TypeName {
                                                        names: [
                                                            Node {
                                                                node: Some(
                                                                    String(
                                                                        String {
                                                                            sval: "percentile_disc_final",
                                                                        },
                                                                    ),
                                                                ),
                                                            },
                                                        ],
                                                        type_oid: 0,
                                                        setof: false,
                                                        pct_type: false,
                                                        typmods: [],
                                                        typemod: -1,
                                                        array_bounds: [],
                                                        location: 185,
                                                    },
                                                ),
                                            ),
                                        },
                                    ),
                                    defaction: DefelemUnspec,
                                    location: 173,
                                },
                            ),
                        ),
                    },
                    Node {
                        node: Some(
                            DefElem(
                                DefElem {
                                    defnamespace: "",
                                    defname: "finalfunc_extra",
                                    arg: None,
                                    defaction: DefelemUnspec,
                                    location: 224,
                                },
                            ),
                        ),
                    },
                ],
                if_not_exists: false,
                replace: true,
            },
        ),
    )
```

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: crates/codegen/src/get_node_properties.rs [hunk 1]
```diff
@@ -559,6 +559,28 @@ fn custom_handlers(node: &Node) -> TokenStream {
...
+                ObjectType::ObjectAggregate => tokens.push(TokenProperty::from(Token::Aggregate)),
+                _ => todo!("Unknown DefineStmt {:#?}", n.kind())
+            }
+            if let Some(node) = &n.args[0].node {
+                if let NodeEnum::List(n) = node {
+                    if n.items.len() == 2 {
+                        tokens.push(TokenProperty::from(Token::Order));
+                        tokens.push(TokenProperty::from(Token::By));
+                        tokens.push(TokenProperty::from(Token::Ident));
+                        tokens.push(TokenProperty::from(Token::Ident));
```
Comment (cvng) [inline crates/codegen/src/get_node_properties.rs:574]: As discussed: Not sure why `Token::Ident` should be pushed twice (or else parser panics)

Comment (cvng) [reply crates/codegen/src/get_node_properties.rs:617]: To reproduce: comment the 3 `tokens.push(TokenProperty::from(Token::Ident))` in `DefineStmt` and run `cargo test valid_statements`

Parser panics twice for the same token with the same span:

```sh
could not find node for token Token { kind: Ident, text: "anyelement", span: 485..495, token_type: NoKeyword } at depth 2
```

**Comment by psteinroe** [comment]
included in #72 - lmk if we can close this @cvng

**Comment by cvng** [comment]
included in #72

**Pull request closed without merging**